contractcourt/utils_test.go

```diff
@@ -24,3 +30,60 @@ func timeout(t *testing.T) func() {
 		close(done)
 	}
 }
+
+func copyFile(dest, src string) error {
+	s, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer s.Close()
+
+	d, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+
+	if _, err := io.Copy(d, s); err != nil {
+		d.Close()
+		return err
+	}
+
+	return d.Close()
+}
+
+// copyChannelState copies the OpenChannel state by copying the database and
+// creating a new struct from it.
+func copyChannelState(state *channeldb.OpenChannel) (
+	*channeldb.OpenChannel, error) {
+
+	// Make a copy of the DB.
+	dbFile := filepath.Join(state.Db.Path(), "channel.db")
+	tempDbPath, err := ioutil.TempDir("", "past-state")
+	if err != nil {
+		return nil, err
+	}
+
+	tempDbFile := filepath.Join(tempDbPath, "channel.db")
+	err = copyFile(tempDbFile, dbFile)
+	if err != nil {
+		return nil, err
+	}
+
+	newDb, err := channeldb.Open(tempDbPath)
```

Comment:
Since `OpenChannel` is mostly a wrapper around access to the underlying DB, only copying the memory state would still lead changes to the original struct making its way to the copied states. We want to "snapshot" the state at this point, so if we don't copy the DB we risk that further modifications will leak into the snapshots.

Added a cleanup method for the temporary files.